en: snapshot-volume backup and restore (#2131)
pingcap/docs-tidb-operator: en/backup-restore-snapshot-perf.md

---
title: Performance of EBS Snapshot Backup and Restore
summary: Learn about the performance of EBS snapshot backup and restore.
---

# Performance of EBS Snapshot Backup and Restore

This document describes the performance of EBS snapshot backup and restore, the factors that affect performance, and the performance test results. The performance metrics are based on the AWS region `us-west-2`.

## Backup performance

> This section introduces the performance of EBS snapshot backup using volumes, the factors that affect performance, and the performance test results. Backup performance is scalable and increases linearly with the number of TiKV nodes.

Author: ```suggestion
This section introduces the performance of EBS snapshot backup using volumes, the factors that affect performance, and the performance test results.
```

### Backup time consumption

> EBS snapshot backup using volumes consists of the following processes: creates a backup task, stops scheduling, stops GC, and obtains the backupts and volume snapshots. For more detailed information about these processes, see [Architecture of EBS snapshot volume backup and restore](volume-snapshot-backup-restore.md). Among these processes, creating a volume snapshot consumes most of the time. Volume snapshots are created in parallel, and the time taken to complete the entire backup task depends on when the most time-consuming volume is created.

Author: ```suggestion
EBS snapshot backup using volumes consists of the following processes: creates a backup task, stops scheduling, disables GC, and obtains the backupts and volume snapshots. For more detailed information about these processes, see [Architecture of EBS snapshot volume backup and restore](volume-snapshot-backup-restore.md). Among these processes, creating a volume snapshot consumes most of the time. Volume snapshots are created in parallel, and the time taken to complete the entire backup task depends on when the most time-consuming volume is created.
```

### Time consumption ratio of backup

| Backup stage     | Time taken    | Total ratio | Remarks                                     |
| :--------: | :---------: | :------: | :-------------------------------------: |

> | Create volume snapshots  | 16 m (50GB) | %99      | Including the time for creating AWS EBS snapshots                  |
> | Others        | 1s          | %1       | Including the time for stopping scheduling, disabling GC, and obtaining the backupts |

Reviewer: ```suggestion
| Create volume snapshots  | 16 m (50 GB) | 99%      | Including the time for creating AWS EBS snapshots                  |
| Others        | 1 s          | 1%       | Including the time for stopping scheduling, disabling GC, and obtaining the backupts |
```

### Backup impact

It is tested that the backup impact on clusters in less than 3% when GP3 volumes are used. In the following figure, the backup is initiated after 10:25.

![EBS Snapshot backup impact](/media/volume-snapshot-backup-impact.jpg)

## Restore performance

This document describes the performance of EBS snapshot restore using volumes, the factors that affect performance, and the performance test results.

### Restore time consumption

EBS snapshot restore using volumes consists of the following processes. For detailed information, see [Architecture of EBS snapshot volume backup and restore](volume-snapshot-backup-restore.md).

1. Create a cluster.

    TiDB Operator creates a cluster in recoveryMode and starts all PD nodes.

2. Restore volumes.

    TiDB Operator creates volume restore subtasks using BR. BR restores the data volumes from the snapshots to start TiKV.

3. Start TiKV.

    TiDB Operator mounts the TiKV volumes and starts TiKV.

> 4. Restores data.

Reviewer: ```suggestion
4. Restore data.
```

    TiDB Operator creates volume data restore subtasks. BR restores the data volumes to a consistent state.

5. Start TiDB.

    TiDB is started and the restore is completed.

### Time consumption ratio of restore

| Restore stage     | Appropriate time taken | Restore ratio | Remarks                                                            |
| :--------: | :---------: | :------: | :-------------------------------------------------------------: |

> | Creates clusters     | 30 s         | %1       | Including the time for downloading docker image and starting PD                                   |
> | Restores volumes     | 20 s         | %1       | Including the time for starting the BR Pod and restoring volumes                                         |

Reviewer: ```suggestion
| Creates clusters     | 30 s         |  1%      | Including the time for downloading docker image and starting PD                                   |
| Restores volumes     | 20 s         |  1%     | Including the time for starting the BR Pod and restoring volumes                                         |
```

> | Starts TiKV   | 2～30 min    | 58%      | If there are ongoing transactions during backup, rocksDB consumes extra 5-30 minutes to start during restore |

Reviewer: 跟中文有点对不上

| 启动 TiKV | 2～30 min | 58% | TiKV 启动受卷性能影响，一般3分钟左右，当遇到有事务的备份做恢复时，此阶段耗时可能达到 30 分钟 |

| Restores data | 2～20 min    | 38%       |   Including the time for restoring data in the raft consensus layer and deleting MVCC data                    |

> | Starts TiDB   | 1 min        | 2%       |  Including the time for download the TiDB docker image and starting TiDB                     |

Reviewer: ```suggestion
| Starts TiDB   | 1 min        | 2%       |  Including the time for downloading the TiDB docker image and starting TiDB                     |
```

### Restore performance data

The duration of volume snapshot restore depends on the time taken to start TiKV and that for data restore. Time taken to start TiKV might be affected by lazy initialization of EBS volume snapshot restore, for which AWS does not provide quantitative metrics. Test data is as follows under the recommended machine type and GP3 storage volume:

![EBS Snapshot restore perf](/media/volume-snapshot-restore-perf.png)

| Volume data  | Total volume size   | Volume configuration             | Appropriate restore duration |
| :------: | :-----: | :---------------: | :--------: |
| 50 GB    | 500 GB  | 7000IOPS/400MiB/s | 16 min    |
| 100 GB   | 500 GB  | 7000IOPS/400MiB/s | 18 min    |
| 200 GB   | 500 GB  | 7000IOPS/400MiB/s | 21 min   |
| 500 GB   | 1024 GB | 7000IOPS/400MiB/s | 25 min   |
| 1024 GB  | 3500 GB | 7000IOPS/400MiB/s | 34 min   |

> **Note:**
>

> > The preceding performance data is for reference only. The actual performance may vary.

Reviewer: 这个提示要不要放到文档开头？
